File: 2279-Maximum-Bags-With-Full-Capacity-of-Rocks/2279-Maximum-Bags-With-Full-Capacity-of-Rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        int count = 0;
        int n = capacity.size();
        vector<int>fullBags(n);
        for (int i = 0; i < n; i++) {
            fullBags[i] = capacity[i] - rocks[i];
            }
        sort(fullBags.begin(), fullBags.end());
        for(auto x: fullBags){
            if(additionalRocks >= x){
                additionalRocks -= x;
                count ++;
            }
        }
        return count;
    }
};

//TC = O(n) because it loops through all the bags once.
//SC = O(1) because it only uses a constant number of variables.

File: 2279-Maximum-Bags-With-Full-Capacity-of-Rocks/2279-Maximum-Bags-With-Full-Capacity-of-Rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        n = len(capacity)
        fullBags = [capacity[i] - rocks[i] for i in range(n)]
        fullBags.sort()
        count = 0
        for x in fullBags:
            if additionalRocks >= x:
                additionalRocks -= x
                count += 1
        return count
